test/e2e/specs/editor/block-computed-styles/paragraph-styles.spec.js

```diff
@@ -0,0 +1,402 @@
+					letterSpacing: '1.3px',
+					lineHeight: '1.3',
+				},
+			};
+
+			await admin.createNewPost();
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Hello',
+					style,
+					fontSize: '13px',
+				},
+			} );
+
+			const block = editor.canvas.getByRole( 'document', {
+				name: 'Block: Paragraph',
+			} );
+
+			// fontStyle, fontWeight, fontSize.
+			await expect( block ).toHaveCSS(
+				'font',
+				'300 12px / 15.6px Times'
+			);
+			await expect( block ).toHaveCSS( 'text-transform', 'capitalize' );
+			await expect( block ).toHaveCSS( 'letter-spacing', '1.3px' );
+			await expect( block ).toHaveCSS(
+				'text-decoration',
+				'line-through solid rgb(0, 0, 0)'
+			);
+			await expect( block ).toHaveCSS( 'line-height', '15.6px' );
+		} );
+	} );
+
+	test( 'Spacing', async ( { admin, editor, requestUtils } ) => {
+		await test.step( 'Global block styles', async () => {
+			const globalBlockStyles = {
+				spacing: {
+					margin: {
+						top: '1px',
+						bottom: '2px',
+					},
+					padding: {
+						top: '3px',
+						right: '4px',
+						bottom: '5px',
+						left: '6px',
+					},
+				},
+			};
+
+			await requestUtils.updateGlobalStyles( {
+				blocks: {
+					'core/paragraph': globalBlockStyles,
+				},
+			} );
+
+			await admin.createNewPost();
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Paragraph 1',
+				},
+			} );
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Paragraph 2',
+				},
+			} );
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: {
+					content: 'Paragraph 3',
+				},
+			} );
+
+			const paragraphs = editor.canvas.getByRole( 'document', {
+				name: 'Block: Paragraph',
+			} );
+
+			// Left/right margin is set to auto by alignments, there's no reliable way to test this.
+			// First block will have its top margin set to zero by the layout.
+			await expect( paragraphs.first() ).toHaveCSS( 'margin-top', '0px' );
+			await expect( paragraphs.first() ).toHaveCSS(
+				'margin-bottom',
+				'2px'
+			);
+			await expect( paragraphs.first() ).toHaveCSS(
+				'padding',
+				'3px 4px 5px 6px'
+			);
+
+			// Second block will have the correct top and bottom margin.
+			await expect( paragraphs.nth( 1 ) ).toHaveCSS(
+				'margin-top',
+				'1px'
+			);
+			await expect( paragraphs.nth( 1 ) ).toHaveCSS(
+				'margin-bottom',
+				'2px'
+			);
+			await expect( paragraphs.nth( 1 ) ).toHaveCSS(
+				'padding',
+				'3px 4px 5px 6px'
+			);
+
+			// Last block will have its bottom margin set to zero by the layout.
+			await expect( paragraphs.last() ).toHaveCSS( 'margin-top', '1px' );
+			await expect( paragraphs.last() ).toHaveCSS(
+				'margin-bottom',
+				'0px'
+			);
+			await expect( paragraphs.last() ).toHaveCSS(
+				'padding',
+				'3px 4px 5px 6px'
+			);
+		} );
+
+		await test.step( 'Block instance styles', async () => {
```

Comment:
For the paragraph block, it might be good to test the padding against a couple of paragraph block instances. That way we can confirm:

- When a background is set on the instance, the padding is overridden as per the `p.has-background` style
- When the instance has background and padding styles, the instance padding styles take precedence

This test would then also cover the [change](https://github.com/WordPress/gutenberg/pull/61638/files#diff-06e5cc378a588f30c09b6d62f89ad8abc373044c2acdf93e20e9ecf3e619aaebL41) to `p.has-background` specificity in https://github.com/WordPress/gutenberg/pull/61638.

Reply:
Good idea, there was already one assertion for the first case in the colors tests, but I moved them to the 'dimensions' section and added an extra assertion for the instance padding overriding the `has-background` padding.